test/class_relationships_test.rb

```diff
@@ -1,77 +1,79 @@
-require 'class_relationships'
+require 'new_impl/class_relationships'
 require 'test_helper'
 
-[
-  [Object, Module, Class],
-  [ToyObject, ToyModule, ToyClass, 'toy_']
-].each do |_Object, _Module, _Class, prefix|
-  describe 'relationships between Object, Module and Class' do
-    include TestHelpers
+module NewImpl
```

Comment:
Putting the whole test inside a module is fine, but the only real advantage is that it allows us to refer to constants defined by that module (`ToyObject`, `ToyModule` and `ToyClass` here) without having to qualify them. Since we only actually refer to those constants once in the entire test, it’s arguably no less convenient to just say `NewImpl::ToyObject`, `NewImpl::ToyModule` and `NewImpl::ToyClass` explicitly and avoid the extra indentation. (If the tests contained a thousand references to those constants then the tradeoff would be different.)